modules/instrument_manager/php/instrumentmanagerprovisioner.class.inc

```diff
@@ -0,0 +1,336 @@
+<?php declare(strict_types=1);
+/**
+ * This file implements a data provisioner to get all modules
+ * for the Module Manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Dave MacFarlane <dave.macfarlane@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\instrument_manager;
+
+/**
+ * This class implements a data provisioner to get all modules
+ * for the module manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Core
+ * @package    Main
+ * @subpackage Core
+ * @author     Dave MacFarlane <dave.macfarlane@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class InstrumentManagerProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Create a ModuleManagerProvisioner, which gets modules for the
+     * module manager menu table.
+     */
+    private \NDB_Factory $_factory;
+
+    /**
+     * The LORIS base path
+     *
+     * @var string
+     */
+    private string $_path;
+     /**
+      * Create a ModuleManagerProvisioner, which gets modules for the
+      * module manager menu table.
+      */
+    function __construct()
+    {
+        $this->_factory = \NDB_Factory::singleton();
+        $this->_path    = $this->_factory->config()->getSetting("base");
+        parent::__construct(
+            "SELECT
+            test_name as Instrument
+	    FROM test_names
+            ",
+            [],
+        );
+    }
+
+    /**
+     * Returns an instance of a HelpRow object for a given
+     * table row.
+     *
+     * @param array $row The database row from the LORIS Database class.
+     *
+     * @return \LORIS\Data\DataInstance An instance representing this row.
+     */
+    public function getInstance($row) : \LORIS\Data\DataInstance
+    {
+        $instrument = $row['Instrument'];
+                $row['Instrument_Type'] = $this->checkInstrumentType($instrument);
+                $row['Table_Installed'] = $this->checkTableInstalled($instrument);
+                $row['Table_Valid']     = $this->checkTableValid($instrument);
+                $row['Pages_Valid']     = $this->checkPagesValid($instrument);
+
+            return new ModuleRow($row);
```

Comment:
This class doesn't seem to serve much point as it's written. Since these are all dynamically calculated properties that are only relevent to the DataInstance, I think it would make more sense to move these functions to the ModuleRow class and organize them into an array in ModuleRow->jsonSerialize(); instead of organizing them into an array here and just having the ModuleRow do nothing. Then the ModuleRow would just be constructed as `new ModuleRow($row['Instrument']);`